### PR TITLE
Add support for new line in text

### LINKED ISSUE #955: Support newlines in symbol text fields
If a text field in a vector tile contains a newline character, it is ignored for rendering. For most geodata this would be uncommon, but I think GL should handle this like Mapnik and render the appropriate line breaks.

screenshot from 2015-01-26 17 17 24

### PR DESCRIPTION
**Pull request by mourner**
Closes #955. I'm not sure this is the cleanest way to fix but it seems to work:

image

 - [x] write tests for all new functionality
 - [x] check benchmarks
 - [x] manually test the debug page
 - [x] get a PR review @ansis @lucaswoj 

cc @ajashton @danpat

### CODE REVIEW

**Comment by mb12** [comment]
@mourner A less intrusive way to fix this would be to handle it the way text transform is implemented.  Basically strip off new lines here. 

https://github.com/mapbox/mapbox-gl-js/blob/master/js/symbol/resolve_text.js#L25

**Comment by mourner** [comment]
@mb12 I don't see how you could implement it there, but maybe I'm missing something — want to do an alternative PR?

File: js/symbol/shaping.js [hunk 1]
```diff
@@ -24,6 +24,8 @@ function Shaping(positionedGlyphs, text, top, bottom, left, right) {
     this.right = right;
 }
 
+var newLine = 0x0a;
```
Comment (1ec5) [inline js/symbol/shaping.js]: What about carriage returns? Can we be sure that Windows-style CRLF won’t occur?

Comment (mourner) [inline js/symbol/shaping.js]: `\r` should be ignored in theory so `\r\n` should work the same way, but I could add a test to make sure the results are the same.

**Comment by ansis** [comment]
Looks good to me!

**Comment by mdehollander** [comment]
@mourner Can you give an example how to use this? I try it on city names via the mapbox editor, but that doesn't seem to get recognized: `{name_zh} \n {name_en}`

**Comment by lanewinfield** [comment]
@mourner Also having issues with this. Has this since been broken?

**Comment by mourner** [comment]
Hmm, I'm not sure. @ChrisLoer @1ec5 were there any changes about how we treat new lines that could break this?

**Comment by ajashton** [comment]
The feature works when editing style JSON directly, but is not supported by the text input in the Studio interface (`\n` is automatically converted to `\\n`). We have an internal ticket open about allowing `\n` via Studio.

**Comment by nextstopsun** [comment]
I'm having issues with this. `\n` in feature properties coming from vector tiles is interpreted as text. No line breaks occur.

Maybe an option `text-newline-character` could be added to the spec to give flexibility in choosing appropriate line break symbol?

image

**Comment by 1ec5** [comment]
@nextstopsun, it sounds like you’re running into this issue even though `\n` is in the vector tiles (as opposed to being input in Studio). Can you open a separate issue so folks can see and triage it? Thanks!

**Pull request merged**